app-frontend/src/app/components/histogram/dataVizHistogram/index.js

```diff
@@ -0,0 +1,114 @@
+import { min, max, range, get } from 'lodash';
+import tpl from './index.html';
+import { Map } from 'immutable';
+
+class DataVizHistogramController {
+    constructor(
+        $rootScope, $scope, $log, $q, $element,
+        graphService
+    ) {
+        'ngInject';
+        $rootScope.autoInject(this, arguments);
+    }
+
+    $postLink() {
+        this.isLoadingHistogram = true;
+        this.$q.all({
+            histStat: this.histogram,
+            histBounds: this.bounds
+        }).then(({histStat, histBounds}) => {
+            const {histogram, statistics} = histStat;
+            if (histogram && histBounds && get(histogram, 'buckets.length')) {
+                this.$scope.$evalAsync(() => this.initGraph(histogram, histBounds));
```

Comment:
I seem to be getting an initialization error on some graphs when loading the view from another view, but not when loading the page fresh with analysis ids

Reply:
@Lknechtli good point. I tested that this can be resolved by doing its following function calls within this `$evalAsync` function instead. It will be included in the last commit.